source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -118,41 +119,30 @@ generated by oVirt:
 
 ![](../../../../images/wiki/upload-ova.png)
 
-1. The front end retrieves the OVF configuration from the OVA.
-2. The front end sends the OVF configuration to the back end.
-3. The back end parses the OVF configuration and returns a VM to the
-   front end.
-4. The front end sends (optionally customized) parameters of the import
-   process to the back end.
-5. The back end persists the VM into the database.
-6. The back end triggers the creation of the target VM disks on the
-   storage domain.
-7. The back end sends the import parameters - including the destination
-   disks - to imageio-proxy, which sends them to imageio.
-8. Imageio-proxy sends the engine a URL to write to. This URL is sent to
-   the front end.
-9. The front end streams the **whole OVA** to imageio-proxy, which
-   streams it to imageio.
-10. Imageio reads the data and according to the import parameters
-    streams the data into the corresponding target disks.
+1. The frontend retrieves the OVF configuration from the OVA.
+2. The frontend sends the OVF configuration to the backend.
+3. The backend parses the OVF configuration and returns a VM object
+   to the frontend.
+4. The frontend sends back the VM (optionally customized) to the backend.
+5. The backend persists the VM into the database and lock it.
+6. The backend sends the list of disks metadata to the frontend,
+   which starts the process of uploading the disks.
+7. From this point the disks upload is identical the regular image
+   upload process.
+8. Upon finishing the last disk upload, releases the VM lock.
 
 Notes:
 
 1. The described process assumes that the disks are not compressed
-   within the OVA. In order to support this, the back end would need to
-   pass an indication to imageio-proxy as to whether or not each disk is
-   compressed (step 7), so the proxy can decompress the data on-the-fly
-   while streaming it to the destination volume (step 10).
-2. In the figure above, the format of the disks within the OVA is QCOW.
-   This will serve as the default format in all OVA files generated by
-   oVirt. However, we should also support the import of OVA files
-   containing RAW format disks.
-3. In the figure above, the destination volumes are in the RAW format.
-   This is the preferable format because these volumes will serve as
-   base volumes, and the RAW format delivers better performance.
-   However, the destination volumes may initially be in the COW format
-   (copied as-is from the OVA file) and only later on converted to the
-   RAW format.
+   within the OVA. In order to support this, the backend would need to
+   pass an indication to the frontend as to whether or not each disk is
+   compressed (step 6), so the ovirt-imageio-daemon can decompress the
+   data on-the-fly while streaming it to the destination volume (step 7).
+2. The upload process supports handle both raw and qcow images in the OVA.
```

Comment:
We support raw sparse, but we cannot preserve sparseness when uploading/downloading files in raw format. The holes in the files are converted to zeros, and libvirt is doing the same when you download a volume. This is why qcow2 format is better for transferring disks.